webapp/src/test/java/io/github/microcks/util/ai/AICopilotHelperTest.java

```diff
@@ -422,4 +427,441 @@ public void testParseRequestResponseOutputSplitWithJson() {
+
+    // Create a ObjectMapper
+    ObjectMapper mapper = new ObjectMapper();
+    // Create a test JsonNode with references and tags
+    JsonNode specNode = mapper.readTree(jsonSpecNode);
+
+    // Call the reduceSpecSize method
+    AICopilotHelper.reduceSpecSize(specNode, "GET /pastry/{name}");
+
+    // Verify that the keys not to keep are removed correctly
+    assertFalse(specNode.has("tags"));
+    assertFalse(specNode.has("components"));
+
+    assertTrue(specNode.has("paths"));
+    assertTrue(specNode.has("openapi"));
+    assertTrue(specNode.has("info"));
+
+    assertTrue(specNode.get("paths").has("/pastry/{name}"));
+    assertFalse(specNode.get("paths").has("/pastry"));
+    // Ensure security property is removed
+    assertFalse(specNode.get("paths").get("/pastry/{name}").has("security"));
+  }
+
+  @Test
+  public void testResolveReferenceAndRemoveTagsInNode() throws Exception {
+
+    // Cadena JSON de ejemplo
```

Comment:
Is it possible to put this in english? Thanks.

Reply:
Offcourse!